craft_application/application.py

```diff
@@ -78,7 +78,7 @@ class AppMetadata:
     project_variables: list[str] = field(default_factory=lambda: ["version"])
     mandatory_adoptable_fields: list[str] = field(default_factory=lambda: ["version"])
 
-    ProjectClass: type[models.Project] = models.Project
+    ProjectClass: type[models.Project] = models.Project  # type: ignore[type-abstract]
```

Comment:
Technically this is a breaking change, since an application that used `models.Project` directly would now not be able to instantiate that project. Given that:

1. This is probably going to require a `craft-application` 3.0
2. Let's do this:

```suggestion
    ProjectClass: type[models.Project]
```

Reply:
Good point.  I dropped the default value.  

This has the annoying side-effect that `ProjectClass` has to be moved up before everything with a default value:

```diff
--- a/craft_application/application.py
+++ b/craft_application/application.py
@@ -70,6 +70,7 @@ class AppMetadata:
     """Metadata about a *craft application."""

     name: str
+    ProjectClass: type[models.Project]
     summary: str | None = None
     version: str = field(init=False)
     source_ignore_patterns: list[str] = field(default_factory=lambda: [])
@@ -78,7 +79,6 @@ class AppMetadata:
     project_variables: list[str] = field(default_factory=lambda: ["version"])
     mandatory_adoptable_fields: list[str] = field(default_factory=lambda: ["version"])

-    ProjectClass: type[models.Project] = models.Project  # type: ignore[type-abstract]
     BuildPlannerClass: type[models.BuildPlanner] = field(
         default=NotImplementedError  # type: ignore[assignment,type-abstract]
     )
```

Reply:
Yeah I'm okay with that anyway since it's a breaking change anyway. Thanks!